src/components/tooltip/Tooltip.js

```diff
@@ -160,6 +169,22 @@ class Tooltip extends React.Component<Props, State> {
         this.fireChildEvent('onKeyDown', event);
     };
 
+    isControlled = () => {
+        const { isShown: isShownProp } = this.props;
+        return typeof isShownProp !== 'undefined';
+    };
+
+    isShown = () => {
+        const { isShown: isShownProp } = this.props;
+        const isControlled = this.isControlled();
+
+        const isShown = isControlled ? isShownProp : this.state.isShown;
```

Comment:
I know this is just a refactor, but this seems a bit strange to me. Should probably implement `getDerivedStateFromProps` or `componentDidUpdate`